DiscordIan/Service/FetchXMLService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using DiscordIan.Model;
+using Microsoft.Extensions.Logging;
+
+namespace DiscordIan.Service
+{
+    public class FetchXMLService
```

Comment:
It looks like a lot of this is a copy of the JSON fetch routine. It would be ideal to abstract this out so the code isn't repeated if possible.